tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaHelper.cs

```diff
@@ -339,11 +339,18 @@ internal static void TryInjectHeaders<TTopicPartitionMarker, TMessage>(
                 if (dataStreamsManager.IsEnabled)
                 {
                     var edgeTags = string.IsNullOrEmpty(topic)
-                        ? defaultProduceEdgeTags
-                        : new[] { "direction:out", $"topic:{topic}", "type:kafka" };
+                                       ? DefaultProduceEdgeTags
+                                       : ["direction:out", $"topic:{topic}", "type:kafka"];
```

Comment:
Not a new issue in this PR, but this line is allocating a new `string[]` and a new string for `$"topic:{topic}"` on every call. Just something to maybe look into later.